### PR TITLE
DEVELOPER-3942 - Added video configurations for exported drupal videos

### PR DESCRIPTION
**Pull request by Dantheman720**
https://issues.jboss.org/browse/DEVELOPER-3942

### CODE REVIEW

**Comment by Dantheman720** [comment]
Hey @unibrew - I think we're good to go on this PR now. If you need me to make any changes, please let me know and I'd be happy to do so!

File: data/provider/jboss-developer.json [hunk 1]
```diff
@@ -370,6 +370,10 @@
       "index": {
         "name": "data_jbossdeveloper_youtube",
         "type": "jbossdeveloper_youtube"
+      },
+      "indexer" : {
+        "type" : "elasticsearch-river-remote",
+        "name" : "jbossdeveloper_books_drupal"
```
Comment (unibrew) [inline data/provider/jboss-developer.json]: prolly copy&paste issue

File: rivers/jbossdeveloper_vimeo_drupal.json [hunk 2]
```diff
@@ -0,0 +1,128 @@
+{
+  "type" : "remote",
+  "remote" : {
+    "urlGetDocuments"               : "http://{apiKey}",
+    "listDocumentsMode"             : "simple",
+    "spacesIndexed"                 : "MAIN",
+    "timeout"                       : "60s",
+    "indexUpdatePeriod"             : "10m",
```
Comment (unibrew) [inline rivers/jbossdeveloper_vimeo_drupal.json]: From yesterday it seems this setting doesn't work, please experiment with the full update one to see if it works as expected. Same for the second river config

**Review by unibrew — COMMENTED** [review]
I just took a quick look but I'll try to make a more detailed one later this week. Please make those changes and update staging so that I can see the data as well. Oh and please email me the REST endpoints for prod, it'll be required for the release.

File: data/provider/jboss-developer.json (hunk 1 above)
Comment (Dantheman720) [reply data/provider/jboss-developer.json]: Oops! Yep. I'll get that fixed.

**Comment by Dantheman720** [comment]
We're running into some issues with the Drupal REST export that is going to prevent this from happening this week.

**Comment by unibrew** [comment]
I took a closer look on the data on staging and the new configs:
- in both youtube and vimeo you have sys_id==sys_content_id which is wrong. However on youtube it's sys_content_id that is wrong, while on vimeo it's 'sys_id' that needs correcting.
- I'm not sure why do you execute this ScriptingPreprocessor to convert 'field_views' and 'field_likes' to JS ints. If those fields are mapped as a number then they will be interpreted, indexed and stored as such.
- sys_created field is not in ISO format, so I'm not sure it'll be parsed correctly and even if, you may get a different timezone than expected. I'd suggest correcting this format on Drupal side otherwise you need to take into consideration Drupal server's timezone location ;)
- Is there any point in having 'full_description' field if the content is same in sys_description ?
- sys_project and target_product are always non-empty arrays and always having one element of an empty string, this looks wrong, there is no such project
- Are you sure that values from source field 'field_video_target_product' will be in sync with 'code' definitions of each sys_project we have https://github.com/searchisko/configuration/tree/master/data/project ? If not then this content cannot be in sys_project and you need to do a mapping to translate the data from 'target_product' to  'sys_project'. If you're sure they will contain the right codes then 'target_product' feels like a duplicate.
- I've noticed that you ignore completely 'path'/'field_video_resource' data from the source, I'm assuming this is intentional as we care only about the RHD url with the video
- 'field_speakers' also ignorable ?
- 'target_product' and 'full_description' are missing mappings but this depends on the above, whether you decide to leave those fields.

**Comment by Dantheman720** [comment]
@unibrew - I made some updates here. 

- I removed the full description since we no longer use it on the site (it was deprecated with the video push).
- I have a PR in for standard ISO format on sys_created (https://github.com/redhat-developer/developers.redhat.com/pull/1894). It will be in UTC once the PR gets merged.
- Target product wasn't initially pushed in with the videos. I imagine that as new videos are added to Drupal, people will be adding target products.
- I removed sys_project as well, because I don't know that the codes will link up. We may need to further discuss this with Paul.
- AFAIK, the field speakers will only show up on the published video pages and won't be used in our search functionality.

**Comment by unibrew** [comment]
I believe we're good to merge this and release to prod. Yesterday you've mentioned that you'd like to do some search tests first. Please let me know when it's ok to release this.

**Comment by Dantheman720** [comment]
@unibrew - I think we're good to go here! :)

**Comment by unibrew** [comment]
released to prod
PS. stats indexes were missing but I've added them myself, you may want to update your fork

**Pull request merged**